apps/tasks/models.py

```diff
@@ -1,45 +1,104 @@
 import datetime
 
+from django.core.mail import send_mail
 from django.db import models
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
 from django.utils import timezone
+from rest_framework.exceptions import ValidationError
 
 from apps.accounts.models import User
 
 
 # Create your models here.
 
+class TaskQuerySet(models.QuerySet):
```

Comment:
Custom manager pentru un annotate e prea jostka, mai bine faci in get_queryset() un annotate